tensorflow_addons/optimizers/tests/rectified_adam_test.py

```diff
@@ -177,3 +177,35 @@ def test_serialization():
     config = tf.keras.optimizers.serialize(optimizer)
     new_optimizer = tf.keras.optimizers.deserialize(config)
     assert new_optimizer.get_config() == optimizer.get_config()
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+def test_schedulers():
+    lr_scheduler = tf.keras.optimizers.schedules.ExponentialDecay(1e-3, 50, 0.5)
+    wd_scheduler = tf.keras.optimizers.schedules.InverseTimeDecay(2e-3, 25, 0.25)
+
+    run_dense_sample(
+        iterations=100,
+        expected=[[0.993192, 1.992625], [2.993369, 3.993239]],
+        optimizer=RectifiedAdam(learning_rate=lr_scheduler, weight_decay=wd_scheduler),
+    )
+
+
+def test_scheduler_serialization():
+    lr_scheduler = tf.keras.optimizers.schedules.ExponentialDecay(1e-3, 50, 0.5)
+    wd_scheduler = tf.keras.optimizers.schedules.InverseTimeDecay(2e-3, 25, 0.25)
+
+    optimizer = RectifiedAdam(learning_rate=lr_scheduler, weight_decay=wd_scheduler)
+    config = tf.keras.optimizers.serialize(optimizer)
+    new_optimizer = tf.keras.optimizers.deserialize(config)
+    assert new_optimizer.get_config() == optimizer.get_config()
+
+    assert new_optimizer.get_config()["learning_rate"] == {
+        "class_name": "ExponentialDecay",
+        "config": lr_scheduler.get_config(),
+    }
+
+    assert new_optimizer.get_config()["weight_decay"] == {
+        "class_name": "InverseTimeDecay",
+        "config": wd_scheduler.get_config(),
+    }
```

Comment:
can we also add tests about feeding `Dict` for `learning_rate` and `weight_decay`?

Reply:
It's already happening in the test, called by this line:
```python
new_optimizer = tf.keras.optimizers.deserialize(config)
````

In fact, adding Dict as input was done to fix a problem caught by the automatic CPU tests. What happens is that if learning rate or weight decay were keras schedulers, calling deserialize actually passes config dicts as inputs for these arguments, in which case you need to call `tf.keras.optimizers.schedules.deserialize`. That's why I had to add Dict.

Reply:
Got it, it makes sense to me.